test/integration/virtualgateway/virtualgateway_test.go

```diff
@@ -361,5 +361,168 @@ var _ = Describe("VirtualGateway", func() {
 			})
 
 		})
+
+		It("Virtual Gateway Connection Pool Scenarios", func() {
```

Comment:
We might need feature tests for outlier detection and connection pools. It's definitely helpful to verify the API part but without feature tests we might've to manually test both the features.

Rest looks straightforward and good to me.

Reply:
Yes, absolutely - goes without saying. The additional traffic check scenarios are planned for e2e tests as follow ups